src/main/java/com/conveyal/taui/controllers/AggregationAreaController.java

```diff
@@ -99,45 +112,69 @@ public static AggregationArea createAggregationArea (Request req, Response res)
 
         ShapefileReader reader = new ShapefileReader(shpFile);
 
-        List<Geometry> geometries = reader.stream().map(f -> (Geometry) f.getDefaultGeometry()).collect(Collectors.toList());
-        UnaryUnionOp union = new UnaryUnionOp(geometries);
-        Geometry merged = union.union();
+        List<SimpleFeature> features = reader.wgs84Stream().collect(Collectors.toList());
+
+        Map<String, Geometry> areas = new HashMap<>();
+
+        if (Boolean.parseBoolean(req.params("union"))) {
```

Comment:
Will this fail if the `union` parameter is missing?

Reply:
I don't think so -- req.params() null if the given param is null or not found and  `Boolean.parseBoolean(null) = false`